.github/workflows/xcframework.yml

```
@@ -0,0 +1,98 @@
+#
+# This source file is part of the Stanford Biodesign Digital Health Group open-source organization
+#
+# SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+#
+# SPDX-License-Identifier: MIT
+#
+
+name: Create XCFramework and Release
+
+on:
+  workflow_call:
+    inputs:
+      workspaceFile:
+        description: 'The path of the Xcode Workspace file (including the file extension).'
+        type: string
+        required: true
+      XCFrameworkName:
+        description: 'The name of the to be created XCFramework. It must not include the default XCFramework extension ".xcframework" as the file naming is automatically done be the workflow.'
+        type: string
+        required: true
+      XCArchiveName:
+        description: 'The name of the to be created XCArchive. It must not include the default XCArchive extension ".xcarchive" as the file naming is automatically done be the workflow.'
+        type: string
+        required: true
+      scheme:
+        description: 'The project scheme that should be used for the archiving process.'
+        type: string
+        required: true
+      version:
+          description: 'The version number of the XCFramework embedded in the XCArchives. This version number is also used for the release tag.'
+          type: string
+          required: true
+      configuration:
+        description: 'The build configuration to use when archiving the scheme, either Debug or Release.'
+        type: string
+        required: false
+        default: 'Release'
+      runsonlabels:
+          description: 'JSON-based collection of labels indicating which type of github runner should be chosen.'
+          required: false
+          type: string
+          default: '["macos-13"]'
+  
+
+jobs:
+  build-xcarchive:
+    name: Build XCArchive
+    uses: ./.github/workflows/archive.yml
+    with:
+      archiveName: ${{ inputs.XCArchiveName }}
+      workspaceFile: ${{ inputs.workspaceFile }}
+      scheme: ${{ inputs.scheme }}
+      version: ${{ inputs.version }}
+      configuration: ${{ inputs.configuration }}
+      runs-on: ${{ fromJson(inputs.runsonlabels) }}
+  create-xcframework-and-release:
+    name: Build XCFramework
+    runs-on: ${{ fromJson(inputs.runsonlabels) }}
+    needs: build-xcarchive
+    steps:
+    - uses: actions/checkout@v3
```

Here is a new v4 out that we should use in all the workflows from now on:
```suggestion
    - uses: actions/checkout@v4
```